### PR TITLE
experiment: Adds Inngest configuration to the experimental cli

### PR DESCRIPTION
**Pull request by dthyresson**
RedwoodJS developers may now discover and configure Inngest via 

```terminal
yarn rw experimental
rw experimental <command>

Run or setup experimental features

Commands:
  rw experimental inngest          Setup Inngest for background, scheduled,
                                   delayed, multi-step, and fan-out jobs
  rw experimental setup <command>  Setup experimental features
  rw experimental studio           Run the Redwood development studio
```

which will

```
✔ Adding Inngest setup packages for RedwoodJS ...
✔ Install inngest packages ...
✔ Configure inngest ...
✔ Add the Inngest GraphQL plugin ...
✔ Add inngest helloWorld example ...
✔ Modify the GraphQL handler to the useInngest plugin ...
✔ Add inngest dev script to package.json
✔ Adding config to redwood.toml...
```

Works in conjunction with https://github.com/inngest/envelop-plugin-inngest/pull/108

A community experiments post will follow as a README.

But TLDR; once configured you can use https://github.com/inngest/envelop-plugin-inngest/tree/main/packages/setup/redwoodjs#function-command to generate Inngest functions.

There is no cli to generate.

### CODE REVIEW

**Comment by dthyresson** [comment]
@thedavidprice  Once merged I can release:

https://github.com/redwoodjs/redwood/pull/8253

and also the hidden Community post https://community.redwoodjs.com/t/ship-background-jobs-crons-webhooks-and-reliable-workflows-in-record-time-with-inngest-and-redwoodjs/4866

File: packages/cli/README.md [hunk 1]
```diff
@@ -111,6 +111,8 @@ cd packages/cli
 yarn dev <command>
 ```
 
+> **Note:** If you are implementing a command and wish to run via `yarn dev <command>` be sure to `yarn build:clean-dist` beforehand to run the latest version.
```
Comment (thedavidprice) [inline packages/cli/README.md:114]: I ❤️ this. Future me thanks you!

**Review by thedavidprice — DISMISSED** [review]
@dthyresson excited to get this in. Thank you for taking the time to add this for discoverability. Dom and I will help push/promote through release communication as a Highlight. 🚀

### TOML
Can you confirm the redwood.toml config added will be:
```
[experimental.inngest]
```

 ### Experimental Setup Discoverability
You got me thinking about how to make "setups" more discoverable as the current command structure was hiding things a level too deep. I'm working on a better, long-term solution. For now, can you make the change in the command name, from:
```
rw experimental inngest
```
...to...
```
rw experimental setup-inngest
```

Note: I'm going to remove `rw experimental setup [command]` via a separate PR.

### Succes Output: add link to forum post for documentation + feedback
Can you add the link to the forum post to display as output once the setup has completed successfully?

See Josh's example over here for setup opentelemetry:
https://github.com/redwoodjs/redwood/blob/456c49a66af6af1fd54a39a2763aec79c9541cfd/packages/cli/src/commands/experimental/setup/opentelemetry/opentelemetryHandler.js#L156C2-L185

File: __fixtures__/example-todo-main/package.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
     "../../packages/*"
   ],
   "devDependencies": {
-    "@redwoodjs/core": "0.32.2"
+    "@redwoodjs/core": "0.32.2",
+    "inngest-setup-redwoodjs": "^0.4.1"
```
Comment (thedavidprice) [inline __fixtures__/example-todo-main/package.json:10]:
```suggestion
    "@redwoodjs/core": "0.32.2"
```

This package shouldn't be added to Fixture. You'll need to revert yarn.lock for fixture as well.

**Comment by dthyresson** [comment]
> Note: I'm going to remove `rw experimental setup [command]` via a separate PR.

Ok, but let's give some thought to the direct organization. Having all the command and handler in one flat experimental directory is going to get messy and confusing.

**Comment by dthyresson** [comment]
> Can you confirm the redwood.toml config added will be:
> 
> ```
> [experimental.inngest]
> ```

Yes. See this test: https://github.com/inngest/envelop-plugin-inngest/blob/2cf10a02f6271eef89abfbc852d7da83d74b2132/packages/setup/redwoodjs/tests/plugin-tasks.test.ts#L137

**Comment by dthyresson** [comment]
> Can you add the link to the forum post to display as output once the setup has completed successfully?

It would be nice to have a lib for repeated experimental tasks:

1. a utility that updates tool to add the experimental flag
2. utility to output some display

**Comment by thedavidprice** [comment]
FYI: https://github.com/redwoodjs/redwood/pull/8256

File: __fixtures__/example-todo-main/package.json (hunk 2 above)
Comment (dthyresson) [reply __fixtures__/example-todo-main/package.json:10]: Also added note in readme that if your command modifies a project, be sure to remove changes in the test project.

**Comment by dthyresson** [comment]
Have added some utils to make setting the epilogue and task end message easier to do:

image


```
/// see util
import chalk from 'chalk'
import terminalLink from 'terminal-link'

const link = (topicId, isTerminal = false) => {
  const communityLink = `https://community.redwoodjs.com/t/${topicId}`
  if (isTerminal) {
    return terminalLink(communityLink, communityLink)
  } else {
    return communityLink
  }
}

export const getEpilogue = (
  command,
  description,
  topicId,
  isTerminal = false
) =>
  `This is an experimental feature to: ${description}.\n\nPlease find documentation and links to provide feedback for ${command} at:\n -> ${link(
    topicId,
    isTerminal
  )}`

export const getTaskEpilogue = (command, description, topicId) => {
  console.log(
    `${chalk.hex('#ff845e')(
      `------------------------------------------------------------------\n 🧪 ${chalk.green(
        'Experimental Feature'
      )} 🧪\n------------------------------------------------------------------`
    )}`
  )
  console.log(getEpilogue(command, description, topicId, false))

  console.log(
    `${chalk.hex('#ff845e')(
      '------------------------------------------------------------------'
    )}\n`
  )
}
```

Uses the command, description and just give it a topicId to generate the output.

Note: I did not update telemetry as it has two messages, but you can easily combine.

**Comment by dthyresson** [comment]
Added some utils as part of the PR for inngest

File: __fixtures__/example-todo-main/package.json (hunk 2 above)
Comment (dthyresson) [reply __fixtures__/example-todo-main/package.json]: Done

**Comment by thedavidprice** [comment]
@Josh-Walker-GM any chance you can help with the review for the utilities? I think it's a great idea — I just can't get to the review until later today.

Thanks @dthyresson!

**Comment by dthyresson** [comment]
Remember all this is in expermental. So all this can change at anytime and can be refactored.

**Comment by dthyresson** [comment]
We’re refactoring the directory structure and even all the comnand names now, so these utils can be chamged, too later

**Comment by dthyresson** [comment]
I just didn’t want to have to add epilogues and tasks for boilerplate messages each time — adn this encourages the posting of a topic to the community as part of implementation

**Review by Josh-Walker-GM — APPROVED** [review]
Yeah I'll not stand in the way with minor edits to util. It looks good, I might have added a function to just return the listr task to be extra lazy but that's a minor refactor that I'll likely want to add when I have to update the existing studio and opentelemetry commands when this goes in.

Just confirming this setup command only works for ts projects?

**Comment by dthyresson** [comment]
Correct. See [https://github.com/inngest/envelop-plugin-inngest/blob/3569318c212a23ea0fdc65a6373219c7bcd62670/packages/setup/redwoodjs/src/index.ts#L46](https://github.com/inngest/envelop-plugin-inngest/blob/3569318c212a23ea0fdc65a6373219c7bcd62670/packages/setup/redwoodjs/src/index.ts#L46) will error if not TypeScript

**Review by thedavidprice — APPROVED** [review]

**Pull request merged**